CHANGELOG.md

```diff
@@ -0,0 +1,22 @@
+# CHANGELOG
```

Comment:
This will be one for @wbthomason to decide but packer hasn't previously maintained a changelog so not sure if we need/should be adding this here since it isn't particularly discoverable. I think this should probably just go in the main docs 🤷🏾

Reply:
I actually favor adding a changelog and linking to it from the main docs; the README is too long as it is.

Reply:
I would prefer to have changes in the changelog be date and commit-stamped, though.

Reply:
To clarify: we also ought to document this feature in the main docs, but I like having a changelog too. Also, the docs should describe what a snapshot does/does not save.